core/embed/rust/src/ui/geometry.rs

```diff
@@ -414,6 +430,16 @@ impl Rect {
             y1: self.y1 + offset.y,
         }
     }
+
+    /// Get all four corner points.
+    pub fn corner_points(&self) -> [Point; 4] {
+        [
+            self.top_left(),
+            self.top_right() - Offset::x(1),
```

Comment:
The reason is the difference between "point" and "edge" - just `top_right` is the right top edge of the `Rect`, but to get the point on the top right, one needs to subtract one pixel on the `x` - similarly for the other two corners.

I had to draw it myself to find this out, because without the Offsets it just does not work as expected

Reply:
Sounds like `top_right`, `bottom_right`, `bottom_left` should be fixed as they return points outside the rectangle?

Reply:
I have "fixed" the `top_right`, `bottom_right` and `bottom_left` and _all_ the UI tests have changes because of this - things like text centering in buttons or scrollbar placing is off because of it.

@mmilata are you OK with this, that probably you will need to fix the centering and other broken stuff on TT?

Reply:
Done in [b3bf5f0](https://github.com/trezor/trezor-firmware/pull/2803/commits/b3bf5f00d6d3ee4ada60ea48769566ea17d61ba1) - you might want to wait for CI to finish the UI tests and see the diff

Reply:
If it changes all UI fixtures and needs some additional work then let's do it outside of this PR.

Reply:
Right, so I will revert the changes from this branch

Reply:
Reverted in [dc60f22](https://github.com/trezor/trezor-firmware/pull/2803/commits/dc60f22f1e2c7ae511a20b99b397e13d5511225f)

Reply:
Removed TODO in [fe85610](https://github.com/trezor/trezor-firmware/pull/2803/commits/fe85610ffe0085843d8fc2de4adf6a78c52e4e82)